docs: Adding info re container signatures (#1355)
containers/skopeo: docs/skopeo-copy.1.md

skopeo  copy --encryption-key jwe:./public.key --encrypt-layer 1 oci:local_nginx:1.17.8 oci:try-encrypt:encrypted
```

## NOTES

In the context of `signatures`, this refers to `container signatures` as implemented by the [github.com/containers/image/signature](https://github.com/containers/image/tree/main/signature) package.

> ```text

Reviewer: I’m not sure we need this quote — just a reference to the `containers-signature(5)` man page might be enough. In particular “the JSON document” seems to me to be mostly irrelevant to users of `skopeo copy`.

Author: Hey @mtrmac yeah, I'd not fully appreciated that these docs are used to generate the `man` pages for `skopeo` and have thus slimmed things down considerably - about to commit ....

> The signature provides an end-to-end authenticated claim that a container image has been approved by a specific party (e.g. the creator of the image as their work, an automated build system as a result of an automated build, a company IT department approving the image for production) under a specified identity (e.g. an OS base image / specific application, with a specific version).

Reviewer 2: nits for consideration
```suggestion
The signature provides an end-to-end authenticated claim that a specific party has approved a container image (e.g., the creator of the image as their work, an automated build system as a result of an automated build, a company IT department approving the image for production) under a specified identity (e.g., an OS base image / specific application, with a specific version).
```

Author: Thanks, as per above comment, I'm slimming down the changes so the verbiage has gone gone gone 🤣

> A container signature consists of a cryptographic signature which identifies and authenticates who signed the image, and carries as a signed payload a JSON document. The JSON document identifies the image being signed, claims a specific identity of the image and if applicable, contains other information about the image.

Reviewer 2: ```suggestion
A container signature consists of a cryptographic signature that identifies and authenticates who signed the image and carries as a signed payload a JSON document. The JSON document identifies the image is signed, claims a specific identity of the image and, if applicable, contains other information about the image.
```

Author: Thanks, as per above comment, I'm slimming down the changes so the verbiage has gone gone gone 🤣
